samza-core/src/main/java/org/apache/samza/operators/impl/PartitionByOperatorImpl.java

```diff
@@ -66,6 +64,8 @@
 
   @Override
   protected void handleInit(Config config, TaskContext context) {
+    this.keyFunction.init(config, context);
```

Comment:
What happens if someone implements these MapFunctions with a TimerFunction? Do we need to tie in the timer callbacks too?

Reply:
PartitionBy should always be triggered by events, not timer. Hence, I don't think there is a use case for TimerFunction as keyFunction. Even there is a use case later, we can always change the implementation of partitionBy() to allow the keyFunction to be a TimerFunction w/o the need to change the API.

Reply:
So the issue is that the keyFunction is just a MapFunction, and users can generally add `implements TimerFunction` to any of their custom functions. So there might be an expectation that this should work, but it will silently do nothing. So either support it, or maybe explicitly disallow this with instanceOf checks at runtime?